jdbcSource/src/test/java/io/vantiq/extsrc/jdbcSource/TestJDBC.java

```diff
@@ -353,229 +352,254 @@ public static void tearDown() throws VantiqSQLException {
     @Test
     public void testProcessPublish() throws VantiqSQLException {
         assumeTrue(testDBUsername != null && testDBPassword != null && testDBURL != null && jdbcDriverLoc != null);
-        jdbc.setupJDBC(testDBURL, testDBUsername, testDBPassword, false, 0);
-        
-        int queryResult;
-        
-        // Try processPublish with a nonsense query
-        try {
-            jdbc.processPublish("jibberish");
-            fail("Should have thrown an exception");
-        } catch (VantiqSQLException e) {
-            // Expected behavior
-        }
-        
-        // Create table that will be used for testing
-        try {
-            queryResult = jdbc.processPublish(CREATE_TABLE);
-            assert queryResult == 0;
-        } catch (VantiqSQLException e) {
-            fail("Should not throw an exception: " + e.getMessage());
-        }
-        
-        // Insert a row of data into the table
         try {
-            queryResult = jdbc.processPublish(PUBLISH_QUERY);
-            assert queryResult > 0;
-        } catch (VantiqSQLException e) {
-            fail("Should not throw an exception: " + e.getMessage());
+            jdbc = new JDBC();
+            jdbc.setupJDBC(testDBURL, testDBUsername, testDBPassword, false, 0);
+    
+            int queryResult;
+    
+            // Try processPublish with a nonsense query
+            try {
+                jdbc.processPublish("jibberish");
+                fail("Should have thrown an exception");
+            } catch (VantiqSQLException e) {
+                // Expected behavior
+            }
+    
+            // Create table that will be used for testing
+            try {
+                queryResult = jdbc.processPublish(CREATE_TABLE);
+                assert queryResult == 0;
+            } catch (VantiqSQLException e) {
+                fail("Should not throw an exception: " + e.getMessage());
+            }
+    
+            // Insert a row of data into the table
+            try {
+                queryResult = jdbc.processPublish(PUBLISH_QUERY);
+                assert queryResult > 0;
+            } catch (VantiqSQLException e) {
+                fail("Should not throw an exception: " + e.getMessage());
+            }
+        } finally {
+            if (jdbc != null) {
+                jdbc.close();
+            }
```

Comment:
It looks like most of the "changes" are just indentation due to this. Could you just do an `@After` instead with this line?

Reply:
Yes -- I thought the async stuff was different, but it's just unused there.  So @Before & @After simplifies things further.